cutevariant/gui/queryrouter.py

```diff
@@ -52,17 +54,17 @@ def widgetChanged(self):
 
         if sender_widget is not None:
             #  update query from sender widget
-            query = sender_widget.getQuery()
+            query = sender_widget.query
             print(query)
 
             if not query:
                 return
 
-            self.query = query
+            self._query = query
 
             print(self.query.sql())
 
             #  change query for all widget except sender
             for widget in self.widgets:
                 if widget != sender_widget:
-                    widget.setQuery(self.query)
+                    widget.query = query
```

Comment:
It may be nice to have  a widget.refresh() method after setting widget.query ?

Reply:
Shouldn't it be the role of the setter ?

Reply:
Yes it is . It just seems strange to me to call an abstract method by calling widget.query = query.